### PR TITLE
Mcts npe fix

### PR DESCRIPTION
**Pull request by koshalt**
Fixes NPE check in deactivation for null subscriptions

### CODE REVIEW

File: api/src/main/java/org/motechproject/nms/api/web/BaseController.java [hunk 1]
```diff
@@ -251,13 +251,45 @@ protected boolean frontLineWorkerAuthorizedForAccess(FrontLineWorker flw, State
...
+    /**
+     * Check if the service is deployed for a given circle
+     * @param service service to check
+     * @param circle circle of the caller
+     * @return true if circle is null/unknown, or
+     * true if no states in circle or
+     * true if one state in circle is authorized, otherwise
+     * false if none of the states in circle are deployed
+     */
+    protected boolean serviceDeployedInCircle(Service service, Circle circle) {
```
Comment (koshalt) [inline api/src/main/java/org/motechproject/nms/api/web/BaseController.java]: No real change here. Moved this method from user controller to base and added comments

File: kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriptionServiceImpl.java [hunk 2]
```diff
@@ -529,8 +529,8 @@ private void deleteCallRetry(String subscriptionId) {
 
     @Override
     public void deactivateSubscription(Subscription subscription, DeactivationReason reason) {
-        if (subscription.getStatus() == SubscriptionStatus.ACTIVE ||
-                subscription.getStatus() == SubscriptionStatus.PENDING_ACTIVATION) {
+        if (subscription != null && (subscription.getStatus() == SubscriptionStatus.ACTIVE ||
```
Comment (koshalt) [inline kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriptionServiceImpl.java]: this is the real change to fix NPE

**Comment by frankhuster** [comment]
No new IT?

**Comment by frankhuster** [comment]
lgtm

**Pull request merged**